sdk/cloudmachine/Azure.Projects/src/Utilities/Polyfill/System.Runtime.CompilerServices.IsExternalInit.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#if !NET5_0_OR_GREATER
+
+using System.ComponentModel;
+namespace System.Runtime.CompilerServices;
+
+[EditorBrowsable(EditorBrowsableState.Never)]
+internal static class IsExternalInit { }
```

Comment:
what is this for?

Reply:
It’s needed for `record` [properties](https://github.com/Azure/azure-sdk-for-net/blob/e68c3081b3094d805736c12f00346a0028115a8c/sdk/cloudmachine/Azure.Projects/src/extensions/BlobExtensions.cs#L44C1-L44C65), like `init`-only properties, in versions of C# before .NET 5.0. This polyfill defines `IsExternalInit` to enable compatibility with older frameworks that don’t support `init` properties.